src/components/Search/Input.tsx

```diff
@@ -57,17 +57,31 @@ const SearchSlash = styled.p`
   }
 `
 
-const Input = ({ query, setQuery, refine, inputRef, ...rest }) => {
+interface IInputProps {
+  query: string
+  setQuery: (arg: string) => void
+  refine: (arg: string) => void
+  inputRef: React.MutableRefObject<HTMLInputElement>
+  [key: string]: unknown
```

Comment:
I have added `[key: string]: unknown` because the Component seems to support other unknown properties as denoted by the usage of the rest operator in the component's signature.

Reply:
Hmm yes, lets not do that. It seems that here we are spreading `styled` props. Lets extend the props with the styled component's props:
```ts
interface IInputProps extends React.ComponentProps<typeof StyledInput> {
  ...
}
```